### ISSUE TITLE
Deprecation warnings during Session creation.

### ISSUE THREAD

**Issue by eirrgang** [radical-cybertools/radical.pilot#2185]
```
# radical-stack 

  python               : 3.6.9
  pythonpath           : 
  virtualenv           : /home/rp/rp-venv

  radical.pilot        : 1.4.1-bb6177a@devel
  radical.saga         : 1.4.0
  radical.utils        : 1.4.0

```

Several DeprecationWarning warnings are issued when calling `radical.pilot.Session()`. They appear to be related to `pymongo` db record interfaces (update, insert, etc).

The DeprecationWarning is not subclassed, to my knowledge, and there are several, with varying messages that aren't easily expressed as a single regular expression. Thus, the warnings cannot be suppressed easily without masking other DeprecationWarnings.

If this issue is closed as "won't fix" because of plans to migrate away from MongoDB, please cite an alternative issue to track.

**Comment by eirrgang** [comment]
Note: There are several other cases that trigger DeprecationWarnings. Most are through the `radical.pilot.db.database` module, but also the `radical.pilot.task_manager` and `radical.pilot.session` modules.

**Comment by mturilli** [comment]
Part of the discussion at https://github.com/radical-cybertools/radical.pilot/discussions/2410

**Comment by mtitov** [comment]
Resolved with new releases (>=1.40), due to dropping the MongoDB requirement
